extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java

```diff
@@ -45,11 +45,7 @@ public void disposeStageSession(@Disposes CompletionStage<Stage.Session> reactiv
     }
 
     public void disposeMutinySession(@Disposes Uni<Mutiny.Session> reactiveSession) {
-        // TODO: @AGG this is not working, need to check w/ Clement to figure out the proper way
-        reactiveSession.on().termination((session, ex, cancelled) -> {
-            if (session != null)
-                session.close();
-        });
+        reactiveSession.subscribe().with(Mutiny.Session::close);
```

Comment:
At this point it could be interesting to figure out if the `reactiveSession` has already been subscribed to, because if not, it will never have been produced and doesn't need to be closed. We might be creating it just to close it.
But I'm pretty sure that the current implementation is based on `CompletionStage` so it's eager and will always be created, so we can forget this optimisation for now.

Reply:
Keep in mind that when you inject the stage version of the session, it will be eagerly created. So the Mutiny version is not less efficient.

I had a look at the session implementation code. It seems that a connection is acquired as soon as the session is created. Perhaps this should be delayed until a connection is actually needed. But I don't know if this is possible in practice.